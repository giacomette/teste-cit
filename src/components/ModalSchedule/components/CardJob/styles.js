import styled from "styled-components";

export const Container = styled.div`
  margin-bottom: 12px;
  cursor: pointer;
`;

export const CardJobId = styled.div`
  width: 65px;
  height: 65px;
  line-height: 2.6;
  text-align: center;
  font-size: 24px;
  background-color: #f7f7f7;
  border-radius: 8px;
`;

export const CardJobTitle = styled.div`
  font-size: 16px;
  margin-bottom: 8px;
`;

export const CardJobLabel = styled.span`
  font-size: 12px;
  color: #666;
`;
