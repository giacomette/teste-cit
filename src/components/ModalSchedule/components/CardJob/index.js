import Grid from "@material-ui/core/Grid";
import { Card, CardContent } from "@material-ui/core";
import moment from "moment";

import { Container, CardJobId, CardJobLabel, CardJobTitle } from "./styles";

function CardJob({ value, onClick }) {
  return (
    <Container onClick={onClick}>
      <Card elevation={2}>
        <CardContent>
          <Grid container spacing={2}>
            <Grid item xs={2}>
              <CardJobId data-testid={`job-id-${value.id}`}>
                {value.id}
              </CardJobId>
            </Grid>
            <Grid item xs={10}>
              <CardJobTitle data-testid={`job-title-${value.id}`}>
                {value.description}
              </CardJobTitle>

              <div>
                <CardJobLabel>
                  Data máxima de conclusão:{" "}
                  <span data-testid={`job-max-execution-date-${value.id}`}>
                    {moment(value.maxExecutionDate).format("DD/MM/YYYY HH:mm")}
                  </span>
                </CardJobLabel>
              </div>

              <div>
                <CardJobLabel data-testid={`job-estimated-time-${value.id}`}>
                  Tempo estimado (Hora): {value.estimatedTime}
                </CardJobLabel>
              </div>
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    </Container>
  );
}

export default CardJob;
