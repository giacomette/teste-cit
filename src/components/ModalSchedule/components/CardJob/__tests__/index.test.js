import { render } from "@testing-library/react";
import CardJob from "../index";

describe("CardJob", () => {
  test("Deve renderizar o card", () => {
    const job = {
      id: 1,
      description: "Importação de arquivos de fundos",
      maxExecutionDate: "2021-02-03 10:20:00",
      estimatedTime: 6,
    };

    render(<CardJob value={job} />);
  });

  test("Deve verificar se houve mudança no card", () => {
    const job = {
      id: 1,
      description: "Importação de arquivos de fundos",
      maxExecutionDate: "2021-02-03 10:20:00",
      estimatedTime: 6,
    };

    const component = render(<CardJob value={job} />);

    expect(component).toMatchSnapshot();
  });

  test("Deve renderizar o titulo do job no card", () => {
    const job = {
      id: 1,
      description: "Importação de arquivos de fundos",
      maxExecutionDate: "2021-02-03 10:20:00",
      estimatedTime: 6,
    };

    const { queryByTestId } = render(<CardJob value={job} />);

    expect(queryByTestId("job-title-1")).not.toBeNull();
  });

  test("Deve renderizar o tempo máximo de conclusão do job no card", () => {
    const job = {
      id: 1,
      description: "Importação de arquivos de fundos",
      maxExecutionDate: "2021-02-03 10:20:00",
      estimatedTime: 6,
    };

    const { queryByTestId } = render(<CardJob value={job} />);

    expect(queryByTestId("job-max-execution-date-1")).not.toBeNull();
  });

  test("Deve renderizar o tempo estimado em horas do job no card", () => {
    const job = {
      id: 1,
      description: "Importação de arquivos de fundos",
      maxExecutionDate: "2021-02-03 10:20:00",
      estimatedTime: 6,
    };

    const { queryByTestId } = render(<CardJob value={job} />);

    expect(queryByTestId("job-estimated-time-1")).not.toBeNull();
  });
});
