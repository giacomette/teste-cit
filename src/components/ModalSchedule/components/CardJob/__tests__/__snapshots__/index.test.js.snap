// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardJob Deve verificar se houve mudança no card 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ ijWURA"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation2 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
              >
                <div
                  class="sc-gsTCUz ibifZU"
                  data-testid="job-id-1"
                >
                  1
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
              >
                <div
                  class="sc-dlfnbm qkDpN"
                  data-testid="job-title-1"
                >
                  Importação de arquivos de fundos
                </div>
                <div>
                  <span
                    class="sc-hKgILt gShUbL"
                  >
                    Data máxima de conclusão:
                     
                    <span
                      data-testid="job-max-execution-date-1"
                    >
                      03/02/2021 10:20
                    </span>
                  </span>
                </div>
                <div>
                  <span
                    class="sc-hKgILt gShUbL"
                    data-testid="job-estimated-time-1"
                  >
                    Tempo estimado (Hora): 
                    6
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ ijWURA"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation2 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <div
                class="sc-gsTCUz ibifZU"
                data-testid="job-id-1"
              >
                1
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
            >
              <div
                class="sc-dlfnbm qkDpN"
                data-testid="job-title-1"
              >
                Importação de arquivos de fundos
              </div>
              <div>
                <span
                  class="sc-hKgILt gShUbL"
                >
                  Data máxima de conclusão:
                   
                  <span
                    data-testid="job-max-execution-date-1"
                  >
                    03/02/2021 10:20
                  </span>
                </span>
              </div>
              <div>
                <span
                  class="sc-hKgILt gShUbL"
                  data-testid="job-estimated-time-1"
                >
                  Tempo estimado (Hora): 
                  6
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
